runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java

```diff
@@ -0,0 +1,294 @@
+	private static final Color HEALTH_COLOR = new Color(225, 35, 0, 125);
+	private static final Color POISONED_COLOR = new Color(0, 145, 0, 150);
+	private static final Color VENOMED_COLOR = new Color(0, 65, 0, 150);
+	private static final Color HEAL_COLOR = new Color(255, 112, 6, 150);
+	private static final Color PRAYER_HEAL_COLOR = new Color(57, 255, 186, 75);
+	private static final Color OVERHEAL_COLOR = new Color(216, 255, 139, 150);
+	private static final int HEIGHT = 252;
+	private static final int RESIZED_BOTTOM_HEIGHT = 272;
+	private static final int WIDTH = 20;
+	private static final int PADDING = 1;
+	private static final int IMAGE_SIZE = 16;
+	private static final int HEALTH_LOCATION_X = 0;
+	private static final int PRAYER_LOCATION_X = 1;
+	private static final int RESIZED_BOTTOM_OFFSET_Y = 12;
+	private static final int RESIZED_BOTTOM_OFFSET_X = 10;
+	private static final int OVERHEAL_OFFSET = 2;
+	private static final int HEAL_OFFSET = 3;
+	private static final int ICON_AND_COUNTER_OFFSET_X = 1;
+	private static final int ICON_AND_COUNTER_OFFSET_Y = 21;
+	private static final int SKILL_ICON_HEIGHT = 35;
+	private static final int COUNTER_ICON_HEIGHT = 18;
+	private static final int OFFSET = 2;
+	private final Client client;
+	private final StatusBarsConfig config;
+	private final SkillIconManager skillIconManager;
+	private final ItemStatChanges itemStatChanges;
+	private final TextComponent textComponent = new TextComponent();
+	private Viewport curViewport;
+	private Widget curWidget;
+
+	@Inject
+	private StatusBarsOverlay(Client client, StatusBarsConfig config, SkillIconManager skillIconManager, ItemStatChanges itemStatChanges)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.client = client;
+		this.config = config;
+		this.skillIconManager = skillIconManager;
+		this.itemStatChanges = itemStatChanges;
+	}
+
+	@Override
+	public Dimension render(Graphics2D g)
+	{
+		for (Viewport viewport : Viewport.values())
+		{
+			Widget viewportWidget = client.getWidget(viewport.getViewport());
+			if (viewportWidget != null && !viewportWidget.isHidden())
+			{
+				curViewport = viewport;
+				curWidget = viewportWidget;
+			}
+		}
+
+		if (curViewport == null || curWidget.isHidden())
+		{
+			return null;
+		}
+
+		Point offsetLeft = curViewport.getOffsetLeft();
+		Point offsetRight = curViewport.getOffsetRight();
+		Point location = curWidget.getCanvasLocation();
+		int offsetPrayerX = (location.getX() - offsetRight.getX());
+		int offsetHealthX = (location.getX() - offsetLeft.getX());
+		int offsetHealthY = (location.getY() - offsetLeft.getY());
+		int offsetPrayerY = (location.getY() - offsetRight.getY());
+		int height = curViewport == Viewport.RESIZED_BOTTOM ? RESIZED_BOTTOM_HEIGHT : HEIGHT;
+
+		if (curViewport == Viewport.RESIZED_BOTTOM)
+		{
+			offsetPrayerY = (location.getY() - RESIZED_BOTTOM_OFFSET_Y - offsetRight.getY());
```

Comment:
Reorder these to be updated in the same order they're assigned above? It's a little weird that they don't have the same ordering.